src/common/graphic-helpers.c

```diff
@@ -57,22 +53,63 @@ multi_rect_set_size(struct multi_rect *rect, int width, int height)
 	 * +-+-----+-+   |
 	 * +---------+  ---
 	 */
+
+	/* Update positions (positions of top/left rectangles are static) */
+	wlr_scene_node_set_position(&rect->top->node,
+		0, 0);
+	wlr_scene_node_set_position(&rect->bottom->node,
+		0, height - line_width);
+	wlr_scene_node_set_position(&rect->left->node,
+		0, line_width);
+	wlr_scene_node_set_position(&rect->right->node,
+		width - line_width, line_width);
+
+	/* Update sizes */
+	wlr_scene_rect_set_size(rect->top,
+		width, line_width);
+	wlr_scene_rect_set_size(rect->bottom,
+		width, line_width);
+	wlr_scene_rect_set_size(rect->left,
+		line_width, height - line_width * 2);
+	wlr_scene_rect_set_size(rect->right,
+		line_width, height - line_width * 2);
+}
+
+static void
+multi_rect_destroy_notify(struct wl_listener *listener, void *data)
+{
+	struct multi_rect *rect = wl_container_of(listener, rect, destroy);
+	for (size_t i = 0; i < 3; i++) {
+		wlr_scene_node_destroy(&rect->outlines[i]->tree->node);
+	}
```

Comment:
Is this necessary? wlr_scene should do this anyway if those are children of the main tree node.